src/v/config/node_config.cc

```diff
@@ -142,6 +142,13 @@ node_config::node_config() noexcept
       "operator intervention is needed to startup the broker.",
       {.visibility = visibility::user},
       std::nullopt)
+  , large_allocation_warning_threshold(
```

Comment:
I think if we make it a cluster property, it becomes a bit too easy to kill a cluster by setting a low enough value to trigger the assertion, then the cluster won't stay alive long enough to talk to the configuration API to reset it.

Reply:
Right now enabling the warning only prints a rate-limited warning log on large assertions. It won't throw any errors. Though it can create a lot of log spam if it's constantly reset to a low value. 

I made it a node config since the only current use case for it is enabling it in CI tests to help us locate large allocations. By making it a node config we avoid the overhead of having it be replicated by raft. It's also a bit of an oddity in the sense that it's a write-only property.  Seastar will dynamically changes the property's value on a shard when the threshold is reached for rate-limiting purposes.